File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/BlackjackConfiguration.java
package com.panopset.blackjackEngine;

import java.util.List;

public interface BlackjackConfiguration extends DeckStacker {
  boolean isDoubleAfterSplitAllowed();
  boolean isResplitAcesAllowed();
  int getSeats();
  int getDecks();
  boolean isDealerHitSoft17();
  boolean isBlackjack6to5();
  boolean isEvenMoneyOnBlackjackVace();
  boolean isLateSurrenderAllowed();
  boolean isEuropeanStyle();
  boolean isFastDeal();
  boolean isBasicStrategyVariationsOnly();
  boolean isShowCount();
  void toggleShowCount();
  List<String> getStrategyData();
  List<String> getCountingSystemData();
  int getLargeBetInWholeDollars();
  int getTargetStakeInWholeDollars();
  int getMinimumBetInWholeDollars();
  int getBetIncrementInWholeDollars();
  int getReloadAmountInWholeDollars();
  boolean isCountVeryPositive();
  boolean isCountVeryNegative();
  boolean isBetIdeaDoubleAfterBust();
  boolean isBetIdeaLetItRideAfterTwoWins();
  BlackjackMessages getMessages();
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/BlackjackConfiguration.kt
package com.panopset.blackjackEngine

import java.util.ArrayList

interface BlackjackConfiguration {
    fun isDoubleAfterSplitAllowed(): Boolean
    fun isResplitAcesAllowed(): Boolean
    fun isSplitAcePlayable(): Boolean
    fun getSeats(): Int
    fun getDecks(): Int
    fun isDealerHitSoft17(): Boolean
    fun isBlackjack6to5(): Boolean
    fun isEvenMoneyOnBlackjackVace(): Boolean
    fun isLateSurrenderAllowed(): Boolean
    fun isEuropeanStyle(): Boolean
    fun isFastDeal(): Boolean
    fun isBasicStrategyVariationsOnly(): Boolean
    fun isShowCount(): Boolean
    fun getStrategyData(): ArrayList<String>
    fun getCountingSystems(): CountingSystems
    fun getLargeBetInWholeDollars(): Int
    fun getTargetStakeInWholeDollars(): Int
    fun getMinimumBetInWholeDollars(): Int
    fun getBetIncrementInWholeDollars(): Int
    fun getReloadAmountInWholeDollars(): Int
    fun getStrategicVeryPositiveCount(): Int
    fun getMessages(): BlackjackMessages
}
